spec/requests/subjects_request_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe "Subjects", type: :request do
+  let(:subject) { FactoryBot.create(:subject) }
+  let(:subjects) { FactoryBot.create_list(:subject, 6) }
+  describe 'GET #index' do
+    before do
+      get subjects_path
+    end
+    it 'index status' do
+      expect(response.status).to eq(200)
+    end
+    it 'render index view' do
+      is_expected.to render_template :index
+    end
+  end
+  describe 'GET #show' do
```

Comment:
add spec with unexisting subject to check 404 error

Reply:
added